contracts/starknet/lib/proposal.cairo

```diff
@@ -2,10 +2,9 @@ from starkware.cairo.common.uint256 import Uint256
 
 struct Proposal {
     quorum: Uint256,
-    snapshot_timestamp: felt,
-    start_timestamp: felt,
-    min_end_timestamp: felt,
-    max_end_timestamp: felt,
+    // timestamps contains the following packed into a single felt (each one is 32 bit):
+    // snapshot_timestamp, start_timestamp, min_end_timestamp, max_end_timestamp
+    timestamps: felt,
```

Comment:
can we use a `TyepAlias` here maybe?

Reply:
how do you mean?

Reply:
the type here is `felt` but we can use `TypeAlias` ([see here](https://www.cairo-lang.org/docs/how_cairo_works/types.html#user-defined-type-aliases)) to have something like:
```
timestamps: PackedTimestamps
```
this way the reader understands that it's a "special" felt :)

Reply:
something like `PackedTimestamps = (packed_timestamps : felt);` seems a little unnecessary. Its an internal value that users wont ever need to interact with directly anyway so i think its okay to keep it a felt.

Reply:
Your choice :) I guess it's just the Rust way of thinking that got me wanting this. I find it much cleaner to have a type wrapper for this `packedTimestamp`. If I'm reading the code and I see this `felt` I will be thinking "what the fuck?". If I read the code and I see it's a special type (that just wraps a felt, but it doesn't matter), I will understand more what it is. But honestly it's a detail :)

Reply:
Im not a fan, I just tried and we would need to do all sorts of casting to make it work. If anything it would make more sense to have a generic type in the FeltUtils lib, a type like `4x32BitPacked`

Reply:
Okay tried doing this and I cant get it to compile. I dont think type aliases can be used for return values

Reply:
lets just stick to felts for the time being I think

Reply:
agreed